frontend/packages/portal-frontend/src/compound/components/DoseResponseTab.tsx

```diff
@@ -256,18 +256,26 @@ class DoseResponseTab extends React.Component<Props, State> {
   }
 
   render() {
+    let doseUnitsXLabel: string | undefined = undefined;
     const columns: Array<LongTableColumn> = this.state.doseResponseTable
       ? Object.keys(this.state.doseResponseTable[0])
           .filter(
             (col) => !["depmapId", "cell_line_display_name"].includes(col)
           )
-          .map((dose) => {
+          .map((doseWithUnits) => {
+            const splitDoseUnits = doseWithUnits.split(/\s/, 2); // units should be separated from dose by one whitespace
+            console.log(splitDoseUnits, doseWithUnits);
+            const dose = splitDoseUnits[0];
+            const units = splitDoseUnits[1];
+            if (doseUnitsXLabel === undefined) {
```

Comment:
Wrote about two concerns about this in a comment which appears to have not gotten saved... so I'll quickly try to rewrite that.

There are two things I'd typically flag as not-ideal: 
1. string parsing to get the units out of the label
2. `map()` is a functional programming idiom, but you have it creating a side effect (mutating `doseUnitsXLabel`) 

I don't believe either of these is a real problem but they are practices which I tend to discourage because they can lead to unexpected consequences when people make changes in the future. I'd say these are a little hacky but ok.

Anyway, there's even more surprising hacks in here already before you touched anything. For example:
```
 ${parseFloat(
              dose.replace("-", ".")	              dose.replace("-", ".")
            ).toPrecision(2)} μM`;	            ).toPrecision(2)} ${units}`            
```

I don't think my concerns are major, just encouraging avoiding those patterns in the future. I think a cleaner solution would be to change `portal-backend/depmap/compound/views/index.py` to include `units` as a separate field rather than append it onto the `dose` field.

Reply:
That's a very good point! I think that passing `units` as a prop rather than a hacky string parsing solution is better for sure. I think I was confused initially because it seemed like this component derived its compound information from datasets rather than having the compound explicitly passed in as a prop but there's no reason why I can't change that!

Reply:
The specific hack you mentioned that existed previously is indeed surprising but there's a note that explains somewhere why that's done on[ this file](https://github.com/broadinstitute/depmap-portal/blob/47284c500ee5389c8de75c952a6ddde6dda15150/portal-backend/depmap/compound/views/index.py#L391)